### PR TITLE
Jwt hw

### PR DESCRIPTION
**Pull request by AnyaRatusnyak**

### CODE REVIEW

File: src/main/java/projectbookstore/config/SecurityConfig.java [hunk 1]
```diff
@@ -33,13 +37,23 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                 .csrf(AbstractHttpConfigurer::disable)
                 .authorizeHttpRequests(
                         auth -> auth
-                                .requestMatchers("/api/auth/**","/swagger-ui/**")
+                                .requestMatchers("/api/auth/**", "/swagger-ui/**")
```
Comment (vovasalyha) [inline src/main/java/projectbookstore/config/SecurityConfig.java]: Add also `/v3/api-docs/**`

File: src/main/java/projectbookstore/dto/UserLoginRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import org.hibernate.validator.constraints.Length;
+
+@Getter
+@Setter
+public class UserLoginRequestDto {
+    @Email
+    @NotBlank
+    private String email;
+    @NotBlank
+    @Length(min = 8, max = 20)
```
Comment (vovasalyha) [inline src/main/java/projectbookstore/dto/UserLoginRequestDto.java]: It's better not to mix `jakarta.validation` and `hibernate.validator` annotations. Better use `Size`

File: src/main/java/projectbookstore/security/JwtUtil.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    public boolean isValidToken(String token) {
+        try {
+            Jws<Claims> claimsJws = Jwts.parserBuilder()
+                    .setSigningKey(secret)
+                    .build()
+                    .parseClaimsJws(token);
+
+            return !claimsJws.getBody().getExpiration().before(new Date());
+        } catch (JwtException | IllegalArgumentException e) {
+            throw new JwtException("Expired or invalid JWT token");
```
Comment (vovasalyha) [inline src/main/java/projectbookstore/security/JwtUtil.java]:
```suggestion
            return false;
```

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -5,3 +5,5 @@ spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 
 spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.show-sql=true
+jwt.expiration=3000000000
+jwt.secret=qwertyuioplkjhgfdsazxcvbnm1234567890
```
Comment (vovasalyha) [inline src/main/resources/application.properties]: It's better to get sensitive properties from system properties and not pushing it to git.

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Looks good, but see a couple of comments.

File: pom.xml [hunk 5]
```diff
@@ -18,6 +18,7 @@
         <maven.checkstyle.plugin.configLocation>checkstyle.xml</maven.checkstyle.plugin.configLocation>
         <lombok.mapstruct.binding.version>0.2.0</lombok.mapstruct.binding.version>
         <mapstruct.version>1.5.5.Final</mapstruct.version>
+        <jjwt.version>0.11.5</jjwt.version>
```
Comment (karma-o) [inline pom.xml:21]:
```suggestion
        <jwt.version>0.11.5</jwt.version>
```

**Review by karma-o — APPROVED** [review]
GJ!

**Pull request merged**